internal/server/app/app.go

```diff
@@ -0,0 +1,138 @@
+package app
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"net"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/m1khal3v/gophkeeper/internal/common/logger"
+	"github.com/m1khal3v/gophkeeper/internal/common/proto"
+	"github.com/m1khal3v/gophkeeper/internal/server/config"
+	grpcs "github.com/m1khal3v/gophkeeper/internal/server/grpc"
+	"github.com/m1khal3v/gophkeeper/internal/server/jwt"
+	"github.com/m1khal3v/gophkeeper/internal/server/manager"
+	"github.com/m1khal3v/gophkeeper/internal/server/repository"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+type App struct {
+	cfg      *config.Config
+	db       *sql.DB
+	services *services
+}
+
+func New() (*App, error) {
+	cfg, err := config.Load()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load config: %w", err)
+	}
+
+	var level string
+	if cfg.Debug {
+		level = "debug"
+	} else {
+		level = "info"
+	}
+	logger.Init("server", level)
+
+	db, err := initDB(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to init db: %w", err)
+	}
+
+	services, err := initServices(db, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to init services: %w", err)
+	}
+
+	return &App{
+		cfg:      cfg,
+		db:       db,
+		services: services,
+	}, nil
+}
+
+func (a *App) Run() error {
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer stop()
+
+	grpcServer := grpc.NewServer(
+		grpc.ChainUnaryInterceptor(
+			grpcs.NewAuthInterceptor(a.services.userManager).Unary(),
+		),
+	)
+	grpcInternal := grpcs.NewServer(a.services.userManager, a.services.dataManager)
+	proto.RegisterAuthServiceServer(grpcServer, grpcInternal)
+	proto.RegisterDataServiceServer(grpcServer, grpcInternal)
+
+	listener, err := net.Listen("tcp", a.cfg.Listen)
+	if err != nil {
+		return fmt.Errorf("failed to create listener: %w", err)
+	}
+
+	go func() {
+		logger.Logger.Info("Starting gRPC server", zap.String("addr", a.cfg.Listen))
+		if err := grpcServer.Serve(listener); err != nil && !errors.Is(err, grpc.ErrServerStopped) {
+			logger.Logger.Error("gRPC server failed", zap.Error(err))
+		}
+
+		stop()
+	}()
+
+	select {
+	case <-ctx.Done():
+		stop()
+	}
+
```

Comment:
зачем такая сложная конструкция

```suggestion
  <-ctx.Done()
	stop()
```